app/models/user.rb

```diff
@@ -738,8 +738,21 @@ def self.system_avatar_template(username)
 
   def self.letter_avatar_color(username)
     username ||= ""
-    color = LetterAvatar::COLORS[Digest::MD5.hexdigest(username)[0...15].to_i(16) % LetterAvatar::COLORS.length]
-    color.map { |c| c.to_s(16).rjust(2, '0') }.join
+    if SiteSetting.restrict_letter_avatar_colors.present?
+      hex_length = 6
+      colors = SiteSetting.restrict_letter_avatar_colors
+      length = colors.length / hex_length
+      color_order_nr = color_order_nr(username, length)
+      index = (color_order_nr * hex_length) + color_order_nr
+      colors[index, hex_length]
+    else
+      color = LetterAvatar::COLORS[color_order_nr(username, LetterAvatar::COLORS.length)]
+      color.map { |c| c.to_s(16).rjust(2, '0') }.join
+    end
+  end
+
+  def self.color_order_nr(username, length)
```

Comment:
Since we're basically computing an index, I think the name of the method should tell that

```suggestion
  def self.color_index(username, length)
```